File: frontend/src/microservices/attraction/objects/BookAttractionItemPlannerRequest.ts
// 本文件定义 attraction 模块的 `BookAttractionItemPlannerRequest`，作为购票请求参数并提供 JSON 编解码。

export type BookAttractionItemPlannerRequest = {
  userId: string
  attractionId: string
  ticketTypeId: string
  sessionId?: string | null
  travelerIds: string[]
  useDate: string
}

export const bookAttractionItemPlannerRequestFromJson = (json: string): BookAttractionItemPlannerRequest =>
  JSON.parse(json) as BookAttractionItemPlannerRequest

export const bookAttractionItemPlannerRequestToJson = (value: BookAttractionItemPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/attraction/objects/BookAttractionItemPlannerRequest.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class BookAttractionItemPlannerRequest(
    userId: String,
    orderId: String,
    attractionId: String,
    ticketTypeId: String,
    sessionId: Option[String],
    travelerIds: List[String],
    useDate: String
)
object BookAttractionItemPlannerRequest:
  given sourceEncoder: Encoder[BookAttractionItemPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[BookAttractionItemPlannerRequest] = deriveDecoder
